src/ObjWriting/Game/T6/AssetDumpers/AssetDumperRawFile.cpp

```diff
@@ -1,12 +1,88 @@
 #include "AssetDumperRawFile.h"
 
+#include <filesystem>
+#include <zlib.h>
+#include <zutil.h>
+
 using namespace T6;
 
+namespace fs = std::filesystem;
+
 bool AssetDumperRawFile::ShouldDump(XAssetInfo<RawFile>* asset)
 {
     return true;
 }
 
+void AssetDumperRawFile::DumpAnimtree(AssetDumpingContext& context, XAssetInfo<RawFile>* asset, std::ostream& stream)
+{
+    const auto* rawFile = asset->Asset();
+
+    if (rawFile->len <= 4)
+    {
+        std::cout << "Invalid len of animtree file \"" << rawFile->name << "\"" << std::endl;
+        return;
+    }
+
+    const auto outLen = reinterpret_cast<const uint32_t*>(rawFile->buffer)[0];
+    const auto inLen = rawFile->len;
+
+    if (outLen > ANIMTREE_MAX_SIZE)
+    {
+        std::cout << "Invalid size of animtree file \"" << rawFile->name << "\": " << outLen << std::endl;
+        return;
+    }
+
+    z_stream_s zs{};
+
+    zs.zalloc = Z_NULL;
+    zs.zfree = Z_NULL;
+    zs.opaque = Z_NULL;
+    zs.avail_in = 0;
+    zs.next_in = Z_NULL;
+
+    int ret = inflateInit2(&zs, -DEF_WBITS);
+
+    if (ret != Z_OK)
+    {
+        throw std::runtime_error("Initializing inflate failed");
+    }
+
+    zs.next_in = reinterpret_cast<const Bytef*>(&rawFile->buffer[4]);
+    zs.avail_in = inLen - sizeof(uint32_t);
+
+    Bytef buffer[0x1000];
+
+    size_t writtenSize = 0;
+    while (zs.avail_in > 0)
+    {
+        zs.next_out = buffer;
+        zs.avail_out = sizeof buffer;
+        ret = inflate(&zs, Z_SYNC_FLUSH);
+
+        if (ret < 0)
+        {
+            std::cout << "Inflate failed for dumping animtree file \"" << rawFile->name << "\"" << std::endl;
+            inflateEnd(&zs);
+            return;
+        }
+
+        const auto inflateOutSize = sizeof buffer - zs.avail_out;
+
+        if (writtenSize + inflateOutSize >= outLen)
+        {
+            // Last byte is a \0 byte. Skip it.
+            stream.write(reinterpret_cast<char*>(buffer), inflateOutSize - 1);
+        }
+        else
+        {
+            stream.write(reinterpret_cast<char*>(buffer), inflateOutSize);
+        }
+        writtenSize += inflateOutSize;
```

Comment:
```suggestion
        const auto inflateOutSize = sizeof buffer - zs.avail_out;
        stream.write(reinterpret_cast<char*>(buffer), inflateOutSize);
        writtenSize += inflateOutSize;
```

I assume you used the code from T5 here. While the last byte being an unnecessary zero byte applies there, it does not seem to be the case for T6 animtrees.

Therefore currently the last character seems to be cut off.

Reply:
Done. Also removed the unused variable writtenSize.